app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt

```diff
@@ -12,24 +14,83 @@ import org.oppia.android.app.model.TopicSummary
 const val DARKEN_VALUE_MULTIPLIER: Float = 0.9f
 const val DARKEN_SATURATION_MULTIPLIER: Float = 1.2f
 
-/** The view model corresponding to topic summaries in the topic summary RecyclerView. */
+/** The view model corresponding to individual topic summaries in the topic summary RecyclerView. */
 class TopicSummaryViewModel(
+  private val activity: AppCompatActivity,
   val topicSummary: TopicSummary,
   val entityType: String,
-  private val topicSummaryClickListener: TopicSummaryClickListener
+  private val topicSummaryClickListener: TopicSummaryClickListener,
+  private val position: Int
 ) : HomeItemViewModel() {
   val name: String = topicSummary.name
   val totalChapterCount: Int = topicSummary.totalChapterCount
+
   @ColorInt
   val backgroundColor: Int = retrieveBackgroundColor()
+
   @ColorInt
   val darkerBackgroundOverlayColor: Int = computeDarkerBackgroundColor()
+  private val marginMax by lazy {
+    activity.resources.getDimensionPixelSize(R.dimen.home_margin_max)
+  }
+  private val marginMin by lazy {
+    activity.resources.getDimensionPixelSize(R.dimen.home_margin_min)
+  }
+  private val spanCount by lazy {
+    activity.resources.getInteger(R.integer.home_span_count)
+  }
 
   /** Callback from data-binding for when the summary tile is clicked. */
   fun clickOnSummaryTile() {
     topicSummaryClickListener.onTopicSummaryClicked(topicSummary)
   }
 
+  fun computeStartMargin(): Int {
+    return when (spanCount) {
+      2 -> when (position % spanCount) {
+        0 -> marginMax
+        else -> marginMin
+      }
+      3 -> when (position % spanCount) {
+        0 -> marginMax
+        1 -> marginMin
+        2 -> 0
+        else -> 0
+      }
+      4 -> when (position % spanCount) {
+        0 -> marginMax
+        1 -> marginMin
+        2 -> marginMin / 2
+        3 -> 0
+        else -> 0
+      }
+      else -> 0
+    }
+  }
+
+  fun computeEndMargin(): Int {
+    return when (spanCount) {
+      2 -> when (position % spanCount) {
+        0 -> marginMin
+        else -> marginMax
+      }
+      3 -> when (position % spanCount) {
+        0 -> 0
+        1 -> marginMin
+        2 -> marginMax
+        else -> 0
+      }
+      4 -> when (position % spanCount) {
+        0 -> 0
+        1 -> marginMin / 2
+        2 -> marginMin
+        3 -> marginMax
+        else -> 0
+      }
+      else -> 0
+    }
+  }
```

Comment:
@rt4914 why are the paddings here different depending on the number of spans? If I understand correctly, this is the computed layout for each case:

https://gist.github.com/BenHenning/0f8181d8bb599be91375bd0fee70f50b.

Why, for cases 3 and 4, are we using smaller between-item paddings than case 1? Also, it seems like we can avoid the 0 cases and simplify this entire computation by always using min/2. If 'min' is meant to be the padding between elements and 'max' is meant to be the padding on the edges of the list, that would also simplify things a lot. Can you provide guidance on why the current setup is the way that it is?

Reply:
@BenHenning I have tried to explain everything in this image, let me know if it makes sense.


![Untitled 001](https://user-images.githubusercontent.com/9396084/102931468-73ae1780-44c4-11eb-962b-4120771d565f.jpeg)

Reply:
Thanks @rt4914. I think the key action item here is to clarify the code. @jcqli can you add a comment explaining that the reason the margins are specifically calculated in this way is because we're limited by the fact that the recycler view is organizing views into columns? We actually can't change the computations to something simpler because we need to fit within the columns (which are always the same size per GridLayoutManager). Rajat's picture above really helps visualize this.

Reply:
Done